doc: Adjust documentation to renamed cilium-sysdump tool

tgraf opened the pull request:
The PR https://github.com/cilium/cilium-sysdump/pull/76 has renamed the cluster-diagnosis tool to
cilium-sysdump and simplified it. Adjust the documentation accordingly.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/10165)
<!-- Reviewable:end -->

rolinh commented on Documentation/troubleshooting.rst:
```diff
@@ -470,13 +422,26 @@ The script has the following list of prerequisites:
+
+.. code:: bash
+
+    $ curl -sLO https://github.com/cilium/cilium-sysdump/releases/download/stable/cilium-sysdump.zip
```
`stable` should actually be replaced by the cilium-sysdump version in the URL.

tgraf replied:
The intent is to the latest stable version assuming that cilium-sysdump is always backwards compatible. Do you see a problem with that?

joestringer replied:
Does github actually provide a notion of a `stable` / latest release zip like this? The URL doesn't work right now.

pchaigno replied:
GitHub doesn't have a `stable` alias, but it does have a `latest`, for the release page only though. If the latest release will always be the stable one for cilium-sysdump, we can do the following:
```bash
# Retrieve ZIP URL from GitHub API:
# We need to -o cilium-sysdump because the last release currently still has the old name.
curl -sL -o cilium-sysdump.zip $(curl -sL https://api.github.com/repos/cilium/cilium-sysdump/releases/latest | jq -r '.assets[].browser_download_url')
# To avoid using jq:
curl -sL -o cilium-sysdump.zip $(curl -sL https://api.github.com/repos/cilium/cilium-sysdump/releases/latest | grep -oP '(?<="browser_download_url": ").*.zip(?=")')
```

aanm replied:
What's the conclusion @tgraf ?

tgraf replied:
According to:
https://help.github.com/en/github/administering-a-repository/linking-to-releases

This will work:
https://github.com/cilium/cilium-sysdump/releases/latest/download/cilium-sysdump.zip

rolinh commented on Documentation/troubleshooting.rst:
```diff
@@ -293,54 +293,6 @@ Opening the ``status``, we can drill down through ``policy.realized.l4``. Do you
-
-* Requires Python >= 2.7.*
-* Requires ``kubectl``.
-* ``kubectl`` should be pointing to your cluster before running the tool.
```
Don't we want to keep the list of requirements in the doc?

tgraf replied:
They are still there, in the section "Automatic log & state collection". The information was duplicated.

joestringer reviewed (approved):
We should also backport this to older branches, right?

tgraf committed:
doc: Adjust documentation to renamed cilium-sysdump tool

The PR cilium/cilium-sysdump/pull/76 has renamed the cluster-diagnosis tool to
cilium-sysdump and simplified it. Adjust the documentation accordingly.

Signed-off-by: Thomas Graf <thomas@cilium.io>

joestringer commented on Documentation/troubleshooting.rst:
```diff
@@ -470,13 +422,26 @@ The script has the following list of prerequisites:
+
+.. code:: bash
+
+    python cilium-sysdump.zip --nodes=$NODE1_IP,$NODE2_IP2
```
nit: These lines don't have `$` but the lines 430-431 do. Can we make it more consistent?

joestringer reviewed (changes requested):
Minor nit below.
